File: Prision/src/main/java/com/gkzxhn/prison/model/iml/SettingModel.java
package com.gkzxhn.prison.model.iml;

import com.android.volley.AuthFailureError;
import com.gkzxhn.prison.async.VolleyUtils;
import com.gkzxhn.prison.common.Constants;
import com.gkzxhn.prison.model.ISettingModel;
import com.gkzxhn.prison.utils.XtHttpUtil;

import org.json.JSONObject;

/**
 * Created by Raleigh.Luo on 18/3/30.
 */

public class SettingModel extends BaseModel implements ISettingModel {
    /**
     * 获取免费呼叫次数
     * @param onFinishedListener
     */
    @Override
    public void requestFreeTime(VolleyUtils.OnFinishedListener<JSONObject> onFinishedListener) {
        try {
            String url=String.format("%s/%s/access_times", Constants.REQUEST_FREE_MEETING_TIME,
                    getSharedPreferences().getString(Constants.USER_ACCOUNT,""));
            volleyUtils.get(JSONObject.class, url,REQUEST_TAG,onFinishedListener);
        } catch (AuthFailureError authFailureError) {
            authFailureError.printStackTrace();
        }
    }
    @Override
    public void requestVersion(VolleyUtils.OnFinishedListener<JSONObject> onFinishedListener) {
        try {
            volleyUtils.get(JSONObject.class,Constants.REQUEST_VERSION_URL,REQUEST_TAG,onFinishedListener);
        } catch (AuthFailureError authFailureError) {
            authFailureError.printStackTrace();
        }
    }
}

File: Prision/src/main/java/com/gkzxhn/prison/model/iml/SettingModel.kt
package com.gkzxhn.prison.model.iml

import com.android.volley.AuthFailureError
import com.gkzxhn.prison.common.Constants
import com.gkzxhn.prison.model.ISettingModel
import com.gkzxhn.prison.utils.XtHttpUtil
import com.gkzxhn.wisdom.async.VolleyUtils

import org.json.JSONObject

/**
 * Created by Raleigh.Luo on 18/3/30.
 */

class SettingModel : CallZijingModel(), ISettingModel {

    /**
     * 获取免费呼叫次数
     * @param onFinishedListener
     */
    override fun requestFreeTime(onFinishedListener: VolleyUtils.OnFinishedListener<JSONObject>) {
        try {
            val url = String.format("%s?terminalNumber=%s", Constants.REQUEST_FREE_MEETING_TIME,
                    sharedPreferences.getString(Constants.USER_ACCOUNT, ""))
            volleyUtils[JSONObject::class.java, url, REQUEST_TAG, onFinishedListener]
        } catch (authFailureError: AuthFailureError) {
            authFailureError.printStackTrace()
        }

    }

    /**
     * 请求版本信息
     */
    override fun requestVersion(onFinishedListener: VolleyUtils.OnFinishedListener<JSONObject>) {
        try {
            val url=String.format("%s?page=1&rows=10",Constants.REQUEST_VERSION_URL)
            volleyUtils[JSONObject::class.java,url, REQUEST_TAG, onFinishedListener]
        } catch (authFailureError: AuthFailureError) {
            authFailureError.printStackTrace()
        }

    }
}
